### ISSUE TITLE
[Feat] Dark mode

### ISSUE THREAD

**Issue by iShawnWang** [papercss/papercss#187]
Since [macOS Mojave 10.14 introduce Drak Mode](https://developer.apple.com/documentation/macos_release_notes/macos_mojave_10_14_release_notes)
It is happy to see paper css also support dark mode, it's awesome ~

**Comment by rhyneav** [comment]
As a dark mode user, this would be awesome. [Here's a good primer](https://css-tricks.com/dark-modes-with-css/) on how this could be introduced. It looks like just a simple media query:

```css
@media (prefers-color-scheme: dark) {

}
``

**Comment by seifsay3d** [comment]
I have recently started to use papercss and I would like to have a dark-mode in my project so I thought that's a nice issue to tackle. 

I have a couple of questions/assumptions that I would like to get clarity on.

1- Do we want to generate paper-dark-min.css and let the user load different builds using `@media (prefers-color-scheme: dark) ` 

2- If not, then I assume we would rely on css variables to dynamically change the variables, then how do we approach IE? would something like this be sufficient?
```
.paper {
  background-color: $main-background;
  background-color: var(--color-main-background);
}
```
  --- 

Currently, I'm thinking of a solution that would be something along these lines

1-  create dark mode variable list with appropriate defaults
```
$dark-mode-primary: #41403e !default;
$dark-mode-secondary: #0071de !default;
```

2- Add sass variables as css variables to html element based on `@media (prefers-color-scheme: dark) ` or html.dark 

3- update all usages for scss vars to css variables with IE fallback

**Comment by rhyneav** [comment]
Hey @seifsay3d sorry for the delay! You bring up a good point about IE (the problem child). Going against my own comment above, I think a solution without using media queries might be the best way to tackle this for now. 

I really like your idea of having a simple `.dark` class that could be put on the `<html>` tag. That way we can leave it up to the user of PaperCSS if they want a dark theme or not. It's easy to make it dynamic for the user's preferences using a few lines of JavaScript. Plus once IE browser support is there, it'll be easy to move this over to using css variables.

So a modified version of your plan would look like:

1. Create dark mode variable list with appropriate defaults
2. Add sass variables as css variables to html element based on `html.dark`
3. Update color references to use some new color mixin (to get light vs dark colors)
4. ???
5. Profit!

What are your thoughts on that?

**Comment by seifsay3d** [comment]
Sounds awesome, I'll give it a shot then 🕺

**Comment by rhyneav** [comment]
That sounds great, thank you for taking it on! Let me know if I can help you in any way!

**Comment by rhyneav** [comment]
Just released in `v1.8.0`!
